ci/ci/build.py

```diff
@@ -89,21 +89,22 @@ def __init__(self, code, config_str, scope, requested_step_names=None):
         for step_config in config['steps']:
             step_params = StepParameters(code, scope, step_config, name_step)
             step = Step.from_json(step_params)
-            self.steps.append(step)
-            name_step[step.name] = step
+            if not step.run_if_requested or step.name in requested_step_names:
+                self.steps.append(step)
+                name_step[step.name] = step
 
-        # transitively close requested_step_names over dependenies
+        # transitively close requested_step_names over dependencies
         if requested_step_names:
             visited = set()
 
             def request(step):
-                if step not in visited:
+                if step and step not in visited:
                     visited.add(step)
                     for s2 in step.deps:
                         request(s2)
 
             for step_name in requested_step_names:
-                request(name_step[step_name])
+                request(name_step.get(step_name))
```

Comment:
I don't think this is necessary: the requested steps should all be present in name_step.

However, this line in Step.__init__:

 >             self.deps = [params.name_step[d] for d in json['dependsOn']]

Will fail if the step depends on a step that is run_if_requested but not requested.